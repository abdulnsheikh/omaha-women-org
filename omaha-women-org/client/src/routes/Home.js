import React from "react";
import OIWC from "../Images/Home.jpg";

export default function Home() {
  return (
    <div className="container">
      <div>
        <img src={OIWC} style={{ width: `${ window.innerWidth < 960 ? window.innerWidth : 960 }px`, height: "auto", margin: '1em'}} />
      </div>

      <div>
        <h5>
        The International Women Club at Omaha (IWC) has been running for 30 years since 1992.
         We are a group of women from the U.S.A. and around the world who share activities that promote friendship and understanding. 
         Every two weeks, we organize various activities about the culture of the United States and different countries. 
         We help women from different countries adapt to life and study in the United States. 
         All of us enjoy different activities, and our purpose is to establish cross-cultural friendships; 
         help immigrants and refugees assimilate; help with, and encourage, learning English; 
         educate on health, climate, safety, education, and recreational issues; and have fun together. 
         Many women have, and continue to, benefit from this club.
        </h5>
      </div>
    </div>
  );
}
