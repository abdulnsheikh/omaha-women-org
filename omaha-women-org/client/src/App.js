import * as React from "react";
import { BrowserRouter, Routes, Route, Link, Navigate } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import Footer from "./Components/footer";
import Header from "./Components/header";
import Events from "./routes/events";
import RegistrationContainer from "./routes/RegistrationContainer";
import AdminDashboard from "./routes/AdminDashboard";
import "./Styles/index.css";
import "./Styles/bootstrap.css";
import LoginPage from "./Components/loginForm";
import Home from "./routes/Home";
import useSessionStorage from "./hooks/useSessionStorage";
import EventRegistrationContainer from "./routes/EventRegister";

export default function App() {
  const [isAuthenticated, toggleAuthenticationFlag] = useSessionStorage("isAuthenticated");
  const [user, setUser] = useSessionStorage("user");
 

  const handleLogout = e => {
    e.preventDefault();
    toggleAuthenticationFlag(false);
    setUser(null);
    window.location = window.location.origin + '/login';
  }

  if (isAuthenticated && user) {
    return (
      <div className="app-container">
        <BrowserRouter>
          <Header user={user} handleLogout={handleLogout}>
            <li className="nav-item" name="home">
              <Link to="/home" className="nav-link">
                Home
              </Link>
            </li>
            <li className="nav-item" name="events">
              <Link to="/events" className="nav-link">
                Events
              </Link>
            </li>
            { user.User_Role === "Admin" && <li className="nav-item">
              <Link to="/admin" className="nav-link" name="dashboard">
                Dashboard
              </Link>
            </li> }
          </Header>
          <Routes>
            <Route path="home" element={<Home />} />
            <Route path="events" element={<Events />} />
            <Route path="admin" element={<AdminDashboard />} />
            <Route path="events/registerEvent" element={<EventRegistrationContainer user={user}/>} />
          </Routes>
          <Footer />
          <ToastContainer />
        </BrowserRouter>
      </div>
    );
  }

  return (
    <div className="app-container">
      <BrowserRouter>
        <Header>
          <li className="nav-item" name="home">
            <Link to="/home" defaultChecked className="nav-link">
              Home
            </Link>
          </li>
          <li className="nav-item" name="events">
            <Link to="/events" className="nav-link">
              Events
            </Link>
          </li>
          <li className="nav-item" name="register">
            <Link to="/register" className="nav-link">
              Register
            </Link>
          </li>
          <li className="nav-item" name="login">
            <Link to="/login" className="nav-link">
              Sign In
            </Link>
          </li>
        </Header>
        <Routes>
          <Route path="/" element={<Navigate replace to="/login" />} />
          <Route path="home" element={<Home />} />
          <Route path="events" element={<Events />} />
          <Route
            path="login"
            element={
              <LoginPage
                toggleAuthenticationFlag={toggleAuthenticationFlag}
                setUser={setUser}
              />
            }
          />
          <Route path="register" element={<RegistrationContainer />} />
          <Route path="events/registerEvent" element={<EventRegistrationContainer user={user}/>} />
        </Routes>
        <Footer />
        <ToastContainer />
      </BrowserRouter>
    </div>
  );
}
