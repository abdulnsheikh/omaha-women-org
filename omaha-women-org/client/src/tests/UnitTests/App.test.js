import { render, screen } from '@testing-library/react';
import App from "../../App.js"
import Home from '../../routes/Home';

describe("Tests for Home Page", () => {
  test("Match snapshot of Home page", () => {
    const { loginContainer } = render(
        <Home />
    );
    expect(loginContainer).toMatchSnapshot();
  });
});


describe("Tests for Guest User", () => {

  const { getByTestId} = render(
    <App />
   );
   const navs = getByTestId("navbar-children");

  test("Match children length of navbar", () => {
    expect(navs.children.length).toBe(4);
  });

});