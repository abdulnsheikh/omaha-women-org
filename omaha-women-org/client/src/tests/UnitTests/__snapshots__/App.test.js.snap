// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Home Page Match snapshot of Home page 1`] = `undefined`;
